bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,425 @@
+            deployment_spec = deployment_pb.spec
+
+            bento_pb = yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    'BentoML currently not support {} repository'.format(
+                        bento_pb.bento.uri.type
+                    )
+                )
+
+            return self._apply(
+                deployment_pb,
+                bento_pb,
+                yatai_service,
+                bento_pb.bento.uri.uri,
+                prev_deployment,
+            )
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
+
+    def _apply(
+        self, deployment_pb, bento_pb, yatai_service, bento_path, prev_deployment
+    ):
+        if loader._is_remote_path(bento_path):
+            with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                return self._apply(
+                    deployment_pb, bento_pb, yatai_service, local_path, prev_deployment
+                )
+
+        try:
+            deployment_spec = deployment_pb.spec
+            aws_config = deployment_spec.aws_lambda_operator_config
+            bento_service_metadata = bento_pb.bento.bento_service_metadata
+
+            py_major, py_minor, _ = bento_service_metadata.env.python_version.split('.')
+            if py_major != '3':
+                raise BentoMLException(
+                    'Python 2 is not supported for Lambda Deployment'
+                )
+            python_runtime = 'python{}.{}'.format(py_major, py_minor)
+
+            api_names = (
+                [aws_config.api_name]
+                if aws_config.api_name
+                else [api.name for api in bento_service_metadata.apis]
+            )
+
+            raise_if_api_names_not_found_in_bento_service_metadata(
+                bento_service_metadata, api_names
+            )
+
+            if prev_deployment:
+                info_json = json.loads(prev_deployment.state.info_json)
+                lambda_s3_bucket = info_json['s3_bucket']
+            else:
+                lambda_s3_bucket = generate_aws_compatible_string(
+                    'btml-{namespace}-{name}-{random_string}'.format(
+                        namespace=deployment_pb.namespace,
+                        name=deployment_pb.name,
+                        random_string=uuid.uuid4().hex[:6].lower(),
+                    )
+                )
+                create_s3_bucket_if_not_exists(lambda_s3_bucket, aws_config.region)
+            deployment_path_prefix = os.path.join(
+                deployment_pb.namespace, deployment_pb.name
+            )
+
+            logger.debug('Uploading artifacts to S3 bucket')
```

Comment:
move line 240-246 to inside `if bento_service_metadata.artifacts:` check

Reply:
make log message more informative:  logger.debug('Uploading artifacts to S3 path: %s/%s', lambda_s3_bucket,artifacts_prefix)